test/unittest/blas1/blas1_rotg_test.cpp

```diff
@@ -26,84 +26,81 @@
+  data_t a = a_input;
+  data_t b = b_input;
+  if (api == api_type::event) {
+    auto device_a = utils::make_quantized_buffer<scalar_t>(ex, a);
+    auto device_b = utils::make_quantized_buffer<scalar_t>(ex, b);
+    auto device_c = utils::make_quantized_buffer<scalar_t>(ex, c);
+    auto device_s = utils::make_quantized_buffer<scalar_t>(ex, s);
+    auto event0 = _rotg(ex, device_a, device_b, device_c, device_s);
+    ex.get_policy_handler().wait(event0);
+
+    auto event1 = ex.get_policy_handler().copy_to_host(device_c, &c, 1);
+    auto event2 = ex.get_policy_handler().copy_to_host(device_s, &s, 1);
+    auto event3 = ex.get_policy_handler().copy_to_host(device_a, &a, 1);
+    auto event4 = ex.get_policy_handler().copy_to_host(device_b, &b, 1);
+    ex.get_policy_handler().wait(event1);
+    ex.get_policy_handler().wait(event2);
+    ex.get_policy_handler().wait(event3);
+    ex.get_policy_handler().wait(event4);
+  }
+  else {
+    _rotg(ex, a, b, c, s);
+  }
+
+  /* When there is an overflow in the calculation of the hypotenuse, results are
+   * implementation defined but r should return inf like reference_blas does */
+  if (std::isinf(a_ref)) {
+    ASSERT_TRUE(std::isinf(a));
+    return;
+  }
+
+  const bool isAlmostEqual = utils::almost_equal<data_t, scalar_t>(a, a_ref) &&
+                             utils::almost_equal<data_t, scalar_t>(b, b_ref) &&
+                             utils::almost_equal<data_t, scalar_t>(c, c_ref) &&
+                             utils::almost_equal<data_t, scalar_t>(s, s_ref);
   ASSERT_TRUE(isAlmostEqual);
 }
 
-#ifdef STRESS_TESTING
-const auto combi =
-    ::testing::Combine(::testing::Values(11, 65, 1002, 1002400),  // size
-                       ::testing::Values(1, 4),                   // incX
-                       ::testing::Values(1, 3)                    // incY
-    );
-#else
-const auto combi = ::testing::Combine(::testing::Values(11, 1002),  // size
-                                      ::testing::Values(4),         // incX
-                                      ::testing::Values(3)          // incY
+/* Using std::numeric_limits<float> to test overflows on float types. If the
+ * test type is something else, then this will be implicitly cast. double should
+ * not overflow */
+const auto combi = ::testing::Combine(
+    ::testing::Values(api_type::event, api_type::result),                  // Api
+    ::testing::Values(0, 2.5, -7.3, std::numeric_limits<float>::max()),    // a
+    ::testing::Values(0, 0.5, -4.3, std::numeric_limits<float>::lowest())  // b
 );
-#endif
 
 template <class T>
 static std::string generate_name(
     const ::testing::TestParamInfo<combination_t<T>>& info) {
-  int size, incX, incY;
-  BLAS_GENERATE_NAME(info.param, size, incX, incY);
+  api_type api;
+  T a, b;
+  BLAS_GENERATE_NAME(info.param, api, a, b);
 }
 
-BLAS_REGISTER_TEST_ALL(Rotg, combination_t, combi, generate_name);
+BLAS_REGISTER_TEST_ALL(Rotg, combination_t, combi, generate_name);
```

Comment:
Missing new line at the end of the file.

Reply:
Fixed